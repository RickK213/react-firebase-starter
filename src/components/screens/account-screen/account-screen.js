import React from 'react';

export const AccountScreen = props => {
  const { authUser } = props;
  const emailLabel = authUser ? authUser.email : '';

  return (
    <div>
      <h2>Account</h2>
      <p>
        <strong>Email Address:</strong>
        <span style={{ marginLeft: '.5rem' }}>{emailLabel}</span>
      </p>
    </div>
  );
};
