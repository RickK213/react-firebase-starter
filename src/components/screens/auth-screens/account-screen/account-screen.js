import React from 'react';
import PropTypes from 'prop-types';
import { AuthUserContext } from '../../../session';
import { PasswordChangeForm } from './password-change-form/password-change-form';

export const AccountScreenComponent = props => {
  const { authUser } = props;
  const emailLabel = authUser ? authUser.email : '';

  return (
    <div>
      <h2>Account</h2>
      <p style={{ marginBottom: '1.5rem' }}>
        <strong>Email Address:</strong>
        <span style={{ marginLeft: '.5rem' }}>{emailLabel}</span>
      </p>
      <p>
        <strong>Change Password:</strong>
      </p>
      <PasswordChangeForm />
    </div>
  );
};

AccountScreenComponent.propTypes = {
  authUser: PropTypes.object
};

AccountScreenComponent.defaultProps = {
  authUser: null
};

export const AccountScreen = () => (
  <div>
    <AuthUserContext.Consumer>
      {authUser => <AccountScreenComponent authUser={authUser} />}
    </AuthUserContext.Consumer>
  </div>
);
