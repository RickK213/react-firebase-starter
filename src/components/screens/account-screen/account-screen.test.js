/* global React, create */
import { AccountScreen } from './account-screen';

describe('<AccountScreen />', () => {
  it('should render correctly', () => {
    const component = create(<AccountScreen />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
