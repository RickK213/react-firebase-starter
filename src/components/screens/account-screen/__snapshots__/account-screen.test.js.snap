// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountScreen /> should render correctly 1`] = `
<div>
  <div>
    <h2>
      Account
    </h2>
    <p
      style={
        Object {
          "marginBottom": "1.5rem",
        }
      }
    >
      <strong>
        Email Address:
      </strong>
      <span
        style={
          Object {
            "marginLeft": ".5rem",
          }
        }
      >
        
      </span>
    </p>
    <p>
      <strong>
        Change Password:
      </strong>
    </p>
    <form
      autoComplete="off"
      onSubmit={[Function]}
      style={
        Object {
          "width": "360px",
        }
      }
    >
      <input
        autoComplete="off"
        name="passwordOne"
        onChange={[Function]}
        placeholder="New Password"
        style={
          Object {
            "display": "block",
            "marginBottom": "1rem",
            "padding": ".5rem",
            "width": "100%",
          }
        }
        type="password"
        value=""
      />
      <input
        autoComplete="off"
        name="passwordTwo"
        onChange={[Function]}
        placeholder="Confirm New Password"
        style={
          Object {
            "display": "block",
            "marginBottom": "1rem",
            "padding": ".5rem",
            "width": "100%",
          }
        }
        type="password"
        value=""
      />
      <button
        disabled={true}
        style={
          Object {
            "padding": ".5rem",
            "textTransform": "uppercase",
          }
        }
        type="submit"
      >
        Change My Password
      </button>
    </form>
  </div>
</div>
`;
